frontend/src/components/ui/WorkspaceFetcher.jsx

```diff
@@ -45,55 +34,82 @@ export default function WorkspaceFetcher() {
     },
     refetchInterval: 2000,
     retry: false,
-    onSuccess: () =>
-      setWorkspace((d) => {
-        d.connected = true;
-      }),
+    onSuccess: () => {
+      if (!workspace.connected) {
+        setWorkspace((d) => {
+          d.connected = true;
+        });
+      }
+    },
   });
-  const getDetails = async (key, existing, configuration, execution) => {
+
+  const syncWorkspace = async (key, existing, configuration, execution) => {
     const fetchedExec = await fetchExecutionDetails(configuration, execution);
     const loadedExecution = await loadExecution(fetchedExec, configuration);
     const isActive = workspace.tabs[key];
     const merged = { ...existing, ...loadedExecution };
     setWorkspace((draft) => {
-      draft.pipelines[key] = merged;
+      draft.tabs[key] = merged;
+    });
+
+    const updatedPipelines = produce(pipelines, (draft) => {
+      draft[key] = merged;
     });
+    setPipelines(updatedPipelines);
     if (isActive) {
       try {
-        await syncResults(key);
+        syncResults(key);
       } catch (err) {
         console.error("Failed to sync results: ", err);
       }
     }
     return merged;
   };
 
+  // Main polling function
+  const queryKey = ["pipelines", configuration?.anvil?.host];
+
+  const {
+    pending,
+    error,
+    data: pipelinesData,
+  } = useQuery({
+    queryKey: queryKey,
+    queryFn: async () => {
+      return await getAllPipelines(configuration);
+    },
+    refetchInterval: 20000,
+    enabled: true,
+  });
+
+  // Polls for executions
   useQueries({
-    queries: Object.keys(workspace.tabs)
+    queries: Object.keys(workspace?.tabs ?? {})
       ?.filter((key) => {
-        const pipeline = workspace.pipelines[key];
+        const pipeline = pipelines[key];
         const existingStatus = pipeline?.record?.Status;
         return existingStatus === "Running" || existingStatus === "Pending";
       })
       .map((key) => {
-        const pipeline = workspace.pipelines[key];
+        const pipeline = pipelines[key];
         const id = key.split(".")[1];
         return {
           queryKey: ["execution", key],
-          queryFn: () => getDetails(key, pipeline, configuration, id),
+          queryFn: () => syncWorkspace(key, pipeline, configuration, id),
           enabled: !pending && !error,
-          refetchInterval: 1000,
+          refetchInterval: 2000,
```

Comment:
shouldn't this be using the `fetchInterval` from the`workspaceAtom`?

Reply:
I removed it because workspaceAtom.fetchInterval doesn't get set by anything